docs/guides/vault-as-secret-store.md

```diff
@@ -0,0 +1,452 @@
+        source: Filesystem
+        fs:
+          path: /vault/secrets/token" | kubectl apply -f -
+```
+
+### Deploy and Test
+
+1. Create a `Composition` and a `CompositeResourceDefinition`:
+
+```shell
+echo "apiVersion: apiextensions.crossplane.io/v1
+kind: CompositeResourceDefinition
+metadata:
+  name: compositeessinstances.ess.example.org
+  annotations:
+    feature: ess
+spec:
+  group: ess.example.org
+  names:
+    kind: CompositeESSInstance
+    plural: compositeessinstances
+  claimNames:
+    kind: ESSInstance
+    plural: essinstances
+  connectionSecretKeys:
+    - publicKey
+    - publicKeyType
+  versions:
+  - name: v1alpha1
+    served: true
+    referenceable: true
+    schema:
+      openAPIV3Schema:
+        type: object
+        properties:
+          spec:
+            type: object
+            properties:
+              parameters:
+                type: object
+                properties:
+                  serviceAccount:
+                    type: string
+                required:
+                  - serviceAccount
+            required:
+              - parameters" | kubectl apply -f -
+              
+echo "apiVersion: apiextensions.crossplane.io/v1
+kind: Composition
+metadata:
+  name: essinstances.ess.example.org
+  labels:
+    feature: ess
+spec:
+  publishConnectionDetailsWithStoreConfigRef: 
+    name: vault
+  compositeTypeRef:
+    apiVersion: ess.example.org/v1alpha1
+    kind: CompositeESSInstance
+  resources:
+    - name: serviceaccount
+      base:
+        apiVersion: iam.gcp.crossplane.io/v1alpha1
+        kind: ServiceAccount
+        metadata:
+          name: ess-test-sa
+        spec:
+          forProvider:
+            displayName: a service account to test ess
+    - name: serviceaccountkey
+      base:
+        apiVersion: iam.gcp.crossplane.io/v1alpha1
+        kind: ServiceAccountKey
+        spec:
+          forProvider:
+            serviceAccountSelector:
+              matchControllerRef: true
+          publishConnectionDetailsTo:
+            name: ess-mr-conn
+            metadata:
+              labels:
+                environment: development
+                team: backend
+            configRef:
+              name: vault
+      connectionDetails:
+        - fromConnectionSecretKey: publicKey
+        - fromConnectionSecretKey: publicKeyType" | kubectl apply -f -
+```
+
+2. Create a `Claim`:
+
+```shell
+echo "apiVersion: ess.example.org/v1alpha1
+kind: ESSInstance
+metadata:
+  name: my-ess
+  namespace: default
+spec:
+  parameters:
+    serviceAccount: ess-test-sa
+  compositionSelector:
+    matchLabels:
+      feature: ess
+  publishConnectionDetailsTo:
+    name: ess-claim-conn
+    metadata:
+      labels:
+        environment: development
+        team: backend
+    configRef:
+      name: vault" | kubectl apply -f -
+```
+
+3. Verify all resources SYNCED and READY:
+
+```shell
+kubectl get managed
+
+kubectl -n default get claim
+
+kubectl get composite
+```
```

Comment:
I got stuck at this point in the guide (this may be because I'm not familiar with Vault at all).

When I created the claim, it wasn't getting READY, even when the managed resources were:
![image](https://user-images.githubusercontent.com/11353912/159378374-71e4aa3a-799a-456d-adf4-71bc64fb7bd9.png)

When I `kubectl describe` the composite, I received this error:
```
Events:
  Type     Reason                   Age                  From                                                             Message
  ----     ------                   ----                 ----                                                             -------
  Normal   SelectComposition        13m (x11 over 18m)   defined/compositeresourcedefinition.apiextensions.crossplane.io  Successfully selected composition
  Warning  ComposeResources         13m (x11 over 18m)   defined/compositeresourcedefinition.apiextensions.crossplane.io  cannot fetch connection secret: cannot fetch connection secret: cannot connect to secret store: cannot extract token: open /vault/secrets/token: no such file or directory
```

Is this path `vault/secrets/token` valid? It seems not to be anywhere else in the guide.

After this error, I tried to delete the claim and the `kubectl delete` was hanging. Should this happen? Also, this event appeared in the XR:
```
  Warning  DeleteCompositeResource  3m9s (x11 over 12m)  defined/compositeresourcedefinition.apiextensions.crossplane.io  cannot unpublish connection details: cannot connect to secret store: cannot extract token: open /vault/secrets/token: no such file or directory
```

Then I needed to delete each the managed resource individually and `kubectl patch` to remove the finalizers from the claim and composite. Is this the desired behavior?

Reply:
Having managed resources as ready is a good indicator meaning it could write its connection secret to Vault. 

For XR/XRC, it looks like the Vault agent didn't get injected as a sidecar which is expected to write the token at `/vault/secrets/token`, and [here](https://www.vaultproject.io/docs/platform/k8s/injector/annotations#vault-hashicorp-com-agent-inject-token) is where that path is coming from. I believe this is expected if you've used latest master build of Crossplane since [this was](https://github.com/crossplane/crossplane/pull/2994) missing, hence required annotations were not added to Crossplane pod to have Vault agent injected and configured.

That change was originally part of this PR but your test reminded me that we need to have it in the release different than the guide itself and opened the PR after seeing your test results.  

So, thanks for testing @Feggah, it really helped a lot 🙌